### ISSUE TITLE
CollectionProperty doesn't propagate delete events through .filter{_ in true}

### ISSUE THREAD

**Issue by iandundas** [ReactiveKit/ReactiveKit#76]
Just noticed this, see the failing test case at the very bottom. When adding an always-true filter to a CollectionProperty, it no longer propagates `delete` events. It does still propagate `update` events, though. 

(This is with v2.0.3)

```swift

import XCTest
import ReactiveKit
import Nimble

class DemonstrateFilterIssueTests: XCTestCase {

    var collection: CollectionProperty<[Cat]>!
    let bag = DisposeBag()
    
    let catA = Cat(value: ["name" : "Mr A"])
    let catB = Cat(value: ["name" : "Mr B"])
    let catC = Cat(value: ["name" : "Mr C"])
    
    
    override func setUp() {
        super.setUp()
        collection = CollectionProperty<[Cat]>([catA, catB])
    }
    
    override func tearDown() {
        bag.dispose()
        collection = nil
        super.tearDown()
    }
    
    func testStartingState() {
        expect(self.collection.count) == 2
    }
    
    func testUpdateWithoutFilter(){
        
        var updates = 0
        collection
            .observeNext { changes in
                updates += changes.deletes.count
            }.disposeIn(bag)
        
        collection.replace([catB, catA], performDiff: true)
        
        expect(updates).toEventually(equal(1))
    }
    
    func testUpdateWithFilter(){
        
        var updates = 0
        collection
            .filter {_ in true} // does nothing (in theory..)
            .observeNext { changes in
                updates += changes.deletes.count
            }.disposeIn(bag)
        
        collection.replace([catB, catA], performDiff: true)
        
        expect(updates).toEventually(equal(1))
    }
    
    func testDeleteWithoutFilter(){
        
        var deletes = 0
        collection
            .observeNext { changes in
                deletes += changes.deletes.count
            }.disposeIn(bag)
        
        collection.replace([catA], performDiff: true)
        
        expect(deletes).toEventually(equal(1))
    }
    
    func testDeleteWithFilter(){
        
        var deletes = 0
        collection
            .filter {_ in true} // does nothing (in theory..)
            .observeNext { changes in
                deletes += changes.deletes.count
            }.disposeIn(bag)
        
        collection.replace([catA], performDiff: true)
        
        expect(deletes).toEventually(equal(1)) // failed: expected to eventually equal <1>, got <0>
    }
}

// demo model
class Cat: Object {
    dynamic var name = ""
    dynamic var miceEaten: Int = 0
    
    dynamic var id: String = NSUUID().UUIDString
    override static func primaryKey() -> String? {
        return "id"
    }
}
```

**Comment by srdanrasic** [comment]
Hah, blame on me for not unit testing those parts of the code! Thanks for reporting this @iandundas!

It should be fixed now in v2.0.4.

**Comment by iandundas** [comment]
thanks! 👌🏻😎
